src/app/static/src/app/components/figures/transformedData.ts

```diff
@@ -6,12 +6,19 @@ export interface TransformationProps {
 }
 
 export function useTransformation(props: TransformationProps) {
-    const evaluateFormula = (formula: string) => {
-        let interpolatedFormula = formula;
-        if (props.settings) {
-            interpolatedFormula = formula.replace(/\{([^}]+)\}/g,
-                (match) => props.settings!![match.replace(/\{|\}/g, "")] as string);
-        }
+    const evaluateFormula = (formula: string, row: any | null = null) => {
+        const interpolatedFormula = formula.replace(/\{([^}]+)\}/g,
+            (match) => {
+                const id = match.replace(/\{|\}/g, "");
+                let val = "";
+                if (props.settings) {
+                    val = props.settings[id] as string;
+                }
+                if (!val && row) {
```

Comment:
So we're relying on settings and column names not to share ids, which is probably fair enough. I suppose an alternative would have been to prefix the values in the formula with its source e.g. settings.population, row.cases_averted, which would make it clear where everything was coming from, but also make it more verbose..

Reply:
Good point and good suggestion also. I'll merge this but open a ticket to future proof it with your suggestion.